### PR TITLE
Searchbar Border

### PR DESCRIPTION
**Pull request by Divya-Haridasula**
Border changes for Searchbar but border not visible

### CODE REVIEW

File: Table_Of_products/src/components/SearchBar.js [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
+
     render() {
         return (
-            <div>
-                <input type="text" value={this.state.searchText} 
-                onInput={this.searchProduct} /> <br/>
-                <input type="checkbox"></input>
+            <div className="searchBarBorder">
+                <input type="text" value={this.state.searchText}
+                    onInput={this.searchProduct} /> <br />
```
Comment (jawad360) [inline Table_Of_products/src/components/SearchBar.js:24]: can you check if we can use `onChange` here https://github.com/facebook/react/issues/3964

File: Table_Of_products/src/components/SearchBar.js [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
+                <p>
+                    <input type="checkbox" onChange={ this.showProductsInStock }></input> <t/>
```
Comment (jawad360) [inline Table_Of_products/src/components/SearchBar.js:26]: you can use something like `<label><input type="checkbox" value="" disabled>Option 3</label>` https://www.w3schools.com/bootstrap/tryit.asp?filename=trybs_form_checkbox&stacked=h

File: Table_Of_products/src/components/SearchBar.js [hunk 3]
```diff
@@ -1,23 +1,32 @@
...
 import PropTypes from 'prop-types';
+import '../App.css';
 
 export default class SearchBar extends Component {
 
     state = {
-        searchText : "",
+        searchText: "",
     }
     searchProduct = (oEvent) => {
```
Comment (jawad360) [inline Table_Of_products/src/components/SearchBar.js:10]: you can use object destructing here `{target: {value}}` and then trim the value

File: Table_Of_products/src/components/SearchBar.js [hunk 4]
```diff
@@ -1,23 +1,32 @@
...
-        let sSearchText= oEvent.target.value.trim();
-        this.setState({searchText : sSearchText});
+        let sSearchText = oEvent.target.value.trim();
+        this.setState({ searchText: sSearchText });
         this.props.searchProduct(sSearchText);
     }
 
+    showProductsInStock = (oEvent)=>{
```
Comment (jawad360) [inline Table_Of_products/src/components/SearchBar.js:16]: use object destructing

File: Table_Of_products/src/App.js [hunk 5]
```diff
@@ -16,29 +16,39 @@ class App extends Component {
...
+    this.setState({ filteredProducts: this.state.Products });
   }
 
   searchProduct = (sSearchText) => {
-    if(sSearchText){
-       this.setState({filteredProducts : this.state.Products.filter((product) => {
-        return (product.category.toLowerCase().includes(sSearchText.toLowerCase()) || product.price.toLowerCase().includes(sSearchText.toLowerCase()) 
+    if (sSearchText) {
+      this.setState({
+        filteredProducts: this.state.Products.filter((product) => {
```
Comment (jawad360) [inline Table_Of_products/src/App.js:29]: no need for return and curly braces if one one statement is preset, can be like
`filteredProducts: this.state.Products.filter((product) => product.category.toLowerCase().includes(sSearchText.toLowerCase()) || product.price.toLowerCase().includes(sSearchText.toLowerCase())
            || product.name.toLowerCase().includes(sSearchText.toLowerCase()));`

File: Table_Of_products/.eslintcache [hunk 6]
```diff
@@ -1 +1 @@
.../React_Apps_Workspace/Table_Of_products/src/components/SearchBar.js",["30"],"import React, { Component } from 'react';\nimport PropTypes from 'prop-types';\n\nexport default class SearchBar extends Component {\n\n    state = {\n        searchText : \"\",\n    }\n    searchProduct = (oEvent) => {\n        let sSearchText= oEvent.target.value.trim();\n        this.setState({searchText : sSearchText});\n        this.props.searchProduct(sSearchText);\n    }\n\n    render() {\n        return (\n            <div>\n                <input type=\"text\" value={this.state.searchText} \n                onInput={this.searchProduct} />\n            </div>\n        );\n    }\n}\n\nlet searchBar = {\n    width: '50px',\n    height: '10px'\n}\n\nSearchBar.propTypes = {\n    searchProduct : PropTypes.func.isRequired\n}\n\n",{"ruleId":"31","replacedBy":"32"},{"ruleId":"33","replacedBy":"34"},{"ruleId":"35","severity":1,"message":"36","line":26,"column":5,"nodeType":"37","messageId":"38","endLine":26,"endColumn":14},"no-native-reassign",["39"],"no-negated-in-lhs",["40"],"no-unused-vars","'searchBar' is assigned a value but never used.","Identifier","unusedVar","no-global-assign","no-unsafe-negation"]
```
Comment (jawad360) [inline Table_Of_products/.eslintcache:1]: add this to gitignore

**Review by jawad360 — CHANGES_REQUESTED** [review]

File: Table_Of_products/src/components/SearchBar.js (hunk 2 above)
Comment (Divya-Haridasula) [reply Table_Of_products/src/components/SearchBar.js:26]: Got it

**Comment by jawad360** [comment]
@Divya-Haridasula please add screenshot as well

File: Table_Of_products/src/components/SearchBar.js (hunk 3 above)
Comment (Divya-Haridasula) [reply Table_Of_products/src/components/SearchBar.js:10]: ok

File: Table_Of_products/src/components/SearchBar.js (hunk 4 above)
Comment (Divya-Haridasula) [reply Table_Of_products/src/components/SearchBar.js:16]: ok

File: Table_Of_products/.eslintcache (hunk 6 above)
Comment (Divya-Haridasula) [reply Table_Of_products/.eslintcache:1]: ok

File: Table_Of_products/src/App.js (hunk 5 above)
Comment (Divya-Haridasula) [reply Table_Of_products/src/App.js:29]: I know this😏

**Pull request merged**